### ISSUE TITLE
I hope Caffe2's python interface add DataLoader Module Similar to pytorch's DataSet and DataLoader

### ISSUE THREAD

**Issue by dongfangduoshou123** [pytorch/pytorch#16254]
I  abstract the trainable cnn model to three parts:DataInput、 convolution blocks、 output/loss。
Train are mostly use python interface for fast implementation， now caffe2 already have many Various operators like conv、bn、losses, and some data read operators;
I find that no matter it is detection, classification, tracking tasks, ---convolution blocks and output/loss are relatively fixed（mostly use Common Structure like resnet50), but The input part of the network is various, especially in the multi-task problems.That is to say, if we want to implement an idea, the most part we should change is the trian input part(which is only used in train mode, inference's input is just an image)， so the trian input part is Unworthy to implement it in cpp my end users.

So I hope Caffe2's python interface could  add DataLoader Module Similar to pytorch's DataSet and DataLoader, so that we could speed up very mush to Realize our ideas。

Thank you!
